core/templates/dev/head/collection_editor/editor_tab/CollectionLinearizerServiceSpec.js

```diff
@@ -0,0 +1,468 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Tests for CollectionLinearizerService.
+ */
+
+describe('Collection linearizer service', function() {
+  var CollectionObjectFactory = null;
+  var CollectionNodeObjectFactory = null;
+  var CollectionLinearizerService = null;
+
+  var firstCollectionNode = null;
+  var secondCollectionNode = null;
+  var thirdCollectionNode = null;
+
+  beforeEach(module('oppia'));
+
+  beforeEach(inject(function($injector) {
+    CollectionObjectFactory = $injector.get('CollectionObjectFactory');
+    CollectionNodeObjectFactory = $injector.get('CollectionNodeObjectFactory');
+    CollectionLinearizerService = $injector.get('CollectionLinearizerService');
+
+    var firstCollectionNodeBackendObject = {
+      exploration_id: 'exp_id0',
+      prerequisite_skills: [],
+      acquired_skills: [],
+      exploration_summary: {
+        title: 'exp title0',
+        category: 'exp category',
+        objective: 'exp objective'
+      }
+    };
+    firstCollectionNode = CollectionNodeObjectFactory.create(
+        firstCollectionNodeBackendObject);
+
+    var secondCollectionNodeBackendObject = {
+      exploration_id: 'exp_id1',
+      prerequisite_skills: [],
+      acquired_skills: [],
+      exploration_summary: {
+        title: 'exp title1',
+        category: 'exp category',
+        objective: 'exp objective'
+      }
+    };
+    secondCollectionNode = CollectionNodeObjectFactory.create(
+        secondCollectionNodeBackendObject);
+
+    var thirdCollectionNodeBackendObject = {
+      exploration_id: 'exp_id2',
+      prerequisite_skills: [],
+      acquired_skills: [],
+      exploration_summary: {
+        title: 'exp title2',
+        category: 'exp category',
+        objective: 'exp objective'
+      }
+    };
+    thirdCollectionNode = CollectionNodeObjectFactory.create(
+        thirdCollectionNodeBackendObject);
+  }));
+
+  var createLinearCollection = function() {
+    var collection = CollectionObjectFactory.createEmptyCollection();
+    firstCollectionNode.getAcquiredSkillList().setSkills(['exp title0']);
+    secondCollectionNode.getPrerequisiteSkillList().setSkills(['exp title0']);
+    secondCollectionNode.getAcquiredSkillList().setSkills(['exp title1']);
+    thirdCollectionNode.getPrerequisiteSkillList().setSkills(['exp title1']);
+    thirdCollectionNode.getAcquiredSkillList().setSkills(['exp title2']);
+
+    // Randomize the add order for robust testing since linearity depends on
+    // skills and not the order in which the nodes where added.
+    collection.addCollectionNode(thirdCollectionNode);
+    collection.addCollectionNode(secondCollectionNode);
+    collection.addCollectionNode(firstCollectionNode);
+    return collection;
+  };
+
+  var createNonLinearCollection = function() {
+    var collection = CollectionObjectFactory.createEmptyCollection();
```

Comment:
Ditto here (having a comment explain the order might be helpful to future maintainers of this test code).